bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/XmlItem.java

```diff
@@ -391,8 +391,8 @@ private boolean isLastChildIsTextNode(List<BXml> childList) {
      */
     @Override
     public XmlValue strip() {
-        children.strip();
-        return this;
+       this.children = (XmlSequence) children.strip();
```

Comment:
Shouldn't this be an empty method, as calling strip on a single element should return it self